datafusion/core/src/datasource/physical_plan/file_scan_config.rs

```diff
@@ -252,19 +257,20 @@ impl DataSource for FileScanConfig {
         // If there is any non-column or alias-carrier expression, Projection should not be removed.
         // This process can be moved into CsvExec, but it would be an overlap of their responsibility.
         Ok(all_alias_free_columns(projection.expr()).then(|| {
-            let mut file_scan = self.clone();
+            let file_scan = self.clone();
```

Comment:
Previously calling `new_exec` *always* cloned. Now it is only cloned when necessary